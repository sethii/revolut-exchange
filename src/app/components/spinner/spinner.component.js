import React from 'react';
import styled from 'react-emotion';
import { keyframes, css } from 'emotion';
import {
  compose,
  branch,
  renderComponent,
  withProps
} from 'recompose';

const Background = styled.div`
  background-color: #048bf2;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const SpinnerWrapper = styled.div`
  width: 40px;
  height: 40px;
  position: relative;
`;

const bounce = keyframes`
  0%, 100% { 
    transform: scale(0.0);
  } 50% { 
    transform: scale(1.0);
  }
`;

const bounceContainer =  css`
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: white;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  
  -webkit-animation: ${bounce} 2.0s infinite ease-in-out;
  animation: ${bounce} 2.0s infinite ease-in-out;
`;

const SpinnerBounce = styled.div`
  composes: ${bounceContainer}
`;

const DelayedSpinnerBounce = styled.div`
  composes: ${bounceContainer};
  animation-delay: -1.0s;
`;

export const Spinner = () => <Background>
  <SpinnerWrapper>
    <SpinnerBounce/>
    <DelayedSpinnerBounce/>
  </SpinnerWrapper>
</Background>;

export const withSpinnerWhileLoading = (checkExpression) => compose(
  withProps(props => ({
    loading: checkExpression(props)
  })),
  branch(
    ({ loading }) => loading,
    renderComponent(Spinner)
  )
);
