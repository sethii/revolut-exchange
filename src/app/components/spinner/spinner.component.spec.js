import React from 'react';
import renderer from 'react-test-renderer';
import { toClass } from 'recompose';

import { Spinner } from './spinner.component';

describe('Spinner component', () => {
  it('renders spinner', () => {
    const SpinnerComponent = toClass(Spinner);
    expect(renderer.create(
      <SpinnerComponent/>
    ).toJSON()).toMatchSnapshot();
  });
});
