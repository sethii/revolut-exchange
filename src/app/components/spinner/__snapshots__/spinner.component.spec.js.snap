// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component renders spinner 1`] = `
.glamor-1 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: white;
  opacity: 0.6;
  position: absolute;
  top: 0px;
  left: 0px;
  -webkit-animation: animation_1db0cce 2.0s infinite ease-in-out;
  animation: animation_1db0cce 2.0s infinite ease-in-out;
}

.glamor-7 {
  background-color: #048bf2;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  position: fixed;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.glamor-5 {
  width: 40px;
  height: 40px;
  position: relative;
}

.glamor-3 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: white;
  opacity: 0.6;
  position: absolute;
  top: 0px;
  left: 0px;
  -webkit-animation: animation_1db0cce 2.0s infinite ease-in-out;
  animation: animation_1db0cce 2.0s infinite ease-in-out;
  -webkit-animation-delay: -1.0s;
  animation-delay: -1.0s;
}

<div
  className="glamor-6 glamor-7"
>
  <div
    className="glamor-4 glamor-5"
  >
    <div
      className="glamor-0 glamor-1"
    />
    <div
      className="glamor-2 glamor-3"
    />
  </div>
</div>
`;
